File: rb/features.rb
# StrictlyBetter SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module StrictlyBetterFeatures
  def self.make_feature(name)
    case name
    when "base"
      StrictlyBetterBaseFeature.new
    when "test"
      StrictlyBetterTestFeature.new
    else
      StrictlyBetterBaseFeature.new
    end
  end
end

File: py/strictlybetter_sdk/features.py
# StrictlyBetter SDK feature factory

from strictlybetter_sdk.feature.base_feature import StrictlyBetterBaseFeature
from strictlybetter_sdk.feature.test_feature import StrictlyBetterTestFeature


def _make_feature(name):
    features = {
        "base": lambda: StrictlyBetterBaseFeature(),
        "test": lambda: StrictlyBetterTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
